0.0.81

Chuchumbrik opened the pull request:
1. Добавлена библиотека DOTween для работы с анимацией
2. Добавлена уровневая система 2.1. В Hero добавлен метод увеличения характеристик в зависимости от уровня 2.2. В Mobs добавлена цена (опыт), за который уровень повышается 2.3. В Observer добавлены переменные и методы для обсчета уровня 2.4. В UIManager Добавлены методы отображения уровня и опыта 2.5. Добавлен ExpSlider
2.5.1 Создан класс ExpSlider для работы с анимацией слайдера exp
3. Мелкие правки

Chuchumbrik committed:
0.0.81

1. Добавлена библиотека DOTween для работы с анимацией
2. Добавлена уровневая система
2.1. В Hero добавлен метод увеличения характеристик в зависимости от уровня
2.2. В Mobs добавлена цена (опыт), за который уровень повышается
2.3. В Observer добавлены переменные и методы для обсчета уровня
2.4. В UIManager Добавлены методы отображения уровня и опыта
2.5. Добавлен ExpSlider
2.5.1 Создан класс ExpSlider для работы с анимацией слайдера exp
3. Мелкие правки

ZagidullinKA commented on Rouge Light Project/Assets/Scripts/Characters/Mobs.cs:
mobsMultipier - опечатка?

Chuchumbrik replied:
Поправил

ZagidullinKA commented on Rouge Light Project/Assets/Scripts/Managers/Observer.cs:
Что обозначает название IncreasetExp?

Chuchumbrik replied:
Увеличение опыта

ZagidullinKA commented on Rouge Light Project/Assets/Scripts/Managers/Observer.cs:
В CheckLvlUp хотелось бы видеть поясняющие логику расчета комментарии

Chuchumbrik replied:
Добавил комменты

Chuchumbrik committed:
0.0.82

1. Поправил название переменной mobMultiplier
2. Добавил комментарии к повышению уровня и анимации
3. Поправил баг повторного убийства врага
3.1. В метод Die добавил уничтожение BoxCollider2D
3.2. Добавил переменную isCanDie в Character - может ли умереть?
3.3. Добавил проверку isCanDie  в переопределенные методы Die для Hero и Mobs

Chuchumbrik committed:
0.0.83

Добавлено отображение урона при попаданию по противнику
1. Создан префаб DamageTextPrefab для отображения текста
2. В метод Awake() Character добавлено определение префаба DamageTextPrefab
3. В метод TakeDamage() Character добавлено:
3.1. Проверка isEnemy
3.2. Создание текста с передачей damage после вычета брони
3.3. Анимация плавного поднятия текст вверх и уменьшение прозрачности до уничтожения объекта
